src/main/java/org/llorllale/cactoos/matchers/MatchesBefore.java

```diff
@@ -49,22 +53,60 @@
  * @param <T> Type of the scalar's value
  * @since 1.0.0
  */
+@SuppressWarnings({
+    "PMD.AvoidCatchingGenericException",
+    "PMD.CallSuperInConstructor",
+    "PMD.ConstructorOnlyInitializesOrCallOtherConstructors"
+    })
 public final class MatchesBefore<T> extends MatcherEnvelope<T> {
+    /**
+     * Time unit.
+     */
+    private static final String UNIT = "milliseconds";
+
     /**
      * Ctor.
      * @param millisec Timeout.
      * @param matcher Matcher.
      */
     public MatchesBefore(final long millisec, final Matcher<T> matcher) {
         super(
-            new MatcherOf<>(
-                new Timed<>(matcher::matches, millisec),
-                desc -> desc
-                    .appendDescriptionOf(matcher)
-                    .appendText(" runs in less than ")
-                    .appendValue(millisec).appendText(" milliseconds"),
-                matcher::describeMismatch
-            )
+            // @checkstyle IllegalCatchCheck (30 lines)
+            // @checkstyle AnonInnerLengthCheck (30 lines)
+            new TypeSafeDiagnosingMatcher<T>() {
```

Comment:
@andreoss what is the problem with using `MatcherOf`, which enforces the implementation of all the needed methods?

Reply:
@victornoel Do we need a proper message on timeout? `MatcherOf` allows only boolean response, for the message we need boolean + mismatch descripton

Reply:
@andreoss I think `MatcherOf` only enforces implementing the various required methods using lambdas passed to its constructor, but you can always manipulate the description in those, no?

Reply:
@victornoel No. In order to have a proper description at timeout we need use `TypeSafeDiagnosingMatcher`, which accepts two arguments in the `match` method, one for item and one for description. The description will be filled in case of `TimeoutException` with proper text. `MatcherOf` does not cover that. We could refactor it to accept `BiFunc<T, Description, Boolean>`

Reply:
@andreoss ok, I better understand, thank you for the explanation. So in that case, let's just have `MatchesBefore` directly implements/extends `TypeSafeDiagnosingMatcher`. There is no need for `MatcherEnvelope` to be used.